### DISCUSSION TITLE
gecko platform

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2911] [Q&A] [answered]
The [docs](https://docs.contiki-ng.org/en/develop/doc/platforms/gecko.html) specify that the brd4162a board is supported.

1) Is an additional programming/main board needed to upload Contiki code onto the brd4162a? How does it connect to the computer?
2) After programming, is an additional board needed to run a network of these devices? How is the brd4162a powered standalone?

Thank you in advance for the information.

BTW, the docs also say that the brd4166a board is supported, but I could not find any information about it, or where can it be bought from.

**Comment by Yagoor** [comment, marked as answer]
Hi.

1- Yes, that is correct. You need the main board to work with the brd4162a: https://www.mouser.dk/ProductDetail/Silicon-Labs/SI-MB4002A?qs=t7xnP681wgUOnj49lvLRXw%3D%3D
2- You don't need the additional board to run the network but you need the main board to power the brd4162a. This board is more for prototyping and testing, it is not for deployment.

If you want a stand alone board the brd4166a would be better but it seems that silicon labs has made it obsolete.

It seems that for the Silicon Labs strategy, they are obsoleting all the series 1 boards and Contiki-NG only has series 1 boards support. 
This is something that will have to be addressed moving forward, so series 2 support needs to be added on Contiki-NG.
I had started to work on it in the past, just never got the time to wrap it up.

**Reply by rahav** [reply]
Thank you @Yagoor for the answer.

Does this mean, in practice, that unless someone intends to build their own mainboard, the Silicon Labs devices are useful only if you pair a [brd4162a](https://www.silabs.com/development-tools/wireless/zigbee/slwrb4162a-efr32mg12-radio-board?tab=overview) with a [Si-MB4002A](https://www.silabs.com/development-tools/wireless/wireless-pro-kit-mainboard?tab=overview)?

I don't think this approach is reasonable unless you get tremendous value from the pair. Such a pair costs $202, so if someone wants to build a testbed with 100 devices, it becomes cumbersome space-wise, adds an unnecessary connection interface, and is expensive. Compared to that, the Nordic architecture with the DK and Dongle is superior.

Do you happen to know if the [SLWSTK6000B EFR32MG Starter Kit](https://www.silabs.com/development-tools/wireless/zigbee/efr32mg-zigbee-thread-starter-kit?tab=overview) is supported by Contiki? It seems to take the same radio board + mainboard paired approach, but it comes with 3 x brd4162a and 3 x mainboards (I think of a different kind).

Your time is a precious resource. In my opinion, it would be preferable to invest it in a modern platform. Is series 2 that?
